lms/djangoapps/instructor/cybersource_enrollment_report.py

```diff
@@ -0,0 +1,130 @@
+                paid_course_reg_item.unit_cost, paid_course_reg_item.line_cost, coupon_codes,
+                paid_course_reg_item.status, paid_course_reg_item.order_id
+            ]
+        else:
+            # check if the user used a registration code for the enrollment.
+            registration_code_redemption = RegistrationCodeRedemption.registration_code_used_for_enrollment(
+                course_enrollment)
+            if registration_code_redemption is not None:
+                registration_code = registration_code_redemption.registration_code
+                if getattr(registration_code, 'invoice_item_id'):
+                    list_price = getattr(registration_code.invoice_item, 'unit_price')
+                    total_amount = registration_code_redemption.registration_code.invoice.total_amount
+                    qty = registration_code_redemption.registration_code.invoice_item.qty
+                    payment_amount = total_amount / qty
+                    is_valid = registration_code_redemption.registration_code.invoice.is_valid
+                    transaction_reference_number = registration_code_redemption.registration_code.invoice_id
+                    if is_valid:
+                        status = _('Invoice Paid')
+                    else:
+                        status = _('Invoice Outstanding')
+
+                    payment_data = [list_price, payment_amount, 'N/A', status, transaction_reference_number]
+                elif getattr(registration_code_redemption.registration_code, 'order_id'):
+                    order_item = OrderItem.objects.get(order=registration_code_redemption.registration_code.order,
+                                                       courseregcodeitem__course_id=course_id)
+
+                    coupon_redemption = CouponRedemption.objects.select_related('coupon').filter(
+                        order_id=registration_code_redemption.registration_code.order)
+                    coupon_codes = [redemption.coupon.code for redemption in coupon_redemption]
+                    coupon_codes = ", ".join(coupon_codes)
+                    payment_data = [
+                        order_item.list_price, order_item.unit_cost, coupon_codes,
+                        order_item.status, order_item.order_id
+                    ]
+                else:
+                    # this happens when the registration code is not created via invoice or bulk purchase
+                    # scenario.
+                    payment_data = [
+                        'N/A', 'N/A', 'N/A', _('Data Integrity Error'), 'N/A'
+                    ]
+            else:
+                # this happens when the user is manually enrolled and
+                # further checks if the user is active or not.
+                if course_enrollment.is_active:
+                    payment_status = _('Manual Entry')
+                else:
+                    payment_status = _('Refunded')
+                payment_data = [
```

Comment:
In this final "else" case, can we just mark payment_status as blank for now. Until we implement the Manual Enrollment tracking SOL-236, we really don't have enough information to really label it as anything definitive.